redbox-core/redbox/models/settings.py

```diff
@@ -19,7 +24,11 @@ class OpenSearchSettings(BaseModel):
 
     model_config = SettingsConfigDict(frozen=True)
 
-    collection_endpoint: str
+    host: str
```

Comment:
@nboyse notice this change, is this ok?

Reply:
Would host be most appropriate or maybe just endpoint? As it also contains creds for DSIT and DBT, but I guess not streamlined with other configs

Reply:
I was thinking something along the lines of this, but maybe too complex for use case:
class OpenSearchSettings(BaseModel):
    """settings required for a aws/opensearch"""

    model_config = SettingsConfigDict(frozen=True)

    collection_endpoint: str = env.str("COLLECTION_ENDPOINT")
    parsed_url: AnyUrl = urlparse(collection_endpoint)

    collection_endpoint__username: Optional[str] = parsed_url.username
    collection_endpoint__password: Optional[str] = parsed_url.password
    collection_endpoint__host: Optional[str] = parsed_url.hostname
    collection_endpoint__port: Optional[str] = parsed_url.port

Reply:
oh i see, in that case how about?

```python
class OpenSearchSettings(BaseModel):
    """settings required for a aws/opensearch"""

    model_config = SettingsConfigDict(frozen=True)
    connection_string: str

    @computed_field
    @property
    def username(self) -> int:
        return urlparse(self.collection_endpoint). username

    @computed_field
    @property
    def password(self) -> int:
        return urlparse(self.collection_endpoint). password

    @computed_field
    @property
    def password(self) -> int:
        return urlparse(self.collection_endpoint). password

    @computed_field
    @property
    def port(self) -> int:
        return urlparse(self.collection_endpoint). port

```